src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,190 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodDriver : GpioDriver
+    {
+        private const string GpioGet = "gpioget";
+        private const string GpioChip0 = "gpiochip0 ";
+        private const string GpioSet = "gpioset";
+        public int _waitMilliSeconds { get; set;}
+        private Dictionary<int, PinMode> _pinNumperToPinModes;
+        private Dictionary<int, Process> _pinNumperToProcess;
+        protected internal override int PinCount => throw new PlatformNotSupportedException("This driver is generic so it can not enumerate how many pins are available.");
+
+        public LibgpiodDriver() {
+            _pinNumperToPinModes = new Dictionary<int, PinMode>();
+            _pinNumperToProcess = new Dictionary<int, Process>();
+            _waitMilliSeconds = (int)TimeSpan.FromSeconds(1).TotalMilliseconds;
+        }
+        protected internal override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventType, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void ClosePin(int pinNumber)
+        {
+            // do nothing_pinNumperToProcess
+        }
+
+        protected internal override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => throw new PlatformNotSupportedException("This driver is generic so it can not perform conversions between pin numbering schemes.");
+
+        protected internal override PinMode GetPinMode(int pinNumber)
+        {
+            if (_pinNumperToPinModes.ContainsKey(pinNumber))
+             {
+                return _pinNumperToPinModes[pinNumber];
+            }
+            else
+            {
+                throw new InvalidOperationException("There was an attempt to get a mode to a pin that is not yet open.");
+            }
+        }
+
+        protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
+        {
+            // Libgpiod driver does not support pull up or pull down resistors.
+            if (mode == PinMode.InputPullDown || mode == PinMode.InputPullUp)
+                return false;
+            return true;
+        }
+
+        protected internal override void OpenPin(int pinNumber)
+        {
+            // do nothing
+        }
+
+        protected internal override PinValue Read(int pinNumber)
+        {
+            PinValue result = default(PinValue);
+            try
+            {
+                Process myProc = new Process();
+                myProc.StartInfo.FileName = GpioGet;
+                myProc.StartInfo.Arguments = $" {GpioChip0} {pinNumber}";
+                myProc.StartInfo.RedirectStandardOutput = true;
+                myProc.Start();
+                if (HandlProcessExit(myProc, $"{GpioGet} {GpioChip0} {pinNumber}", _waitMilliSeconds))
```

Comment:
Sounds like you want a method instead that is named something like `WaitForProcessToExit` and also you should have some code that handles the else in case this method returns false meaning that the process didn't exit in time or that the exit code wasn't 0.